### PR TITLE
[SNOW-916052] Use python None instead of empty string for tombstone e2e tests

### PR DESCRIPTION
**Pull request by sfc-gh-rcheng**
Snowflake converters and community converters convert empty string records different. Snowflake converters treat "" as a valid record, community converters treat "" as a tombstone record. Use python None (null) instead of empty string to accurately test tombstone ingestion

Discussed offline with @sfc-gh-tzhang to use None instead

### CODE REVIEW

File: src/main/java/com/snowflake/kafka/connector/records/RecordService.java [hunk 1]
```diff
@@ -566,7 +566,6 @@ public boolean shouldSkipNullValue(
       // get valueSchema
       Schema valueSchema = record.valueSchema();
       if (valueSchema instanceof SnowflakeJsonSchema) {
-        // TODO SNOW-916052: will not skip if record.value() == null
```
Comment (sfc-gh-rcheng) [inline src/main/java/com/snowflake/kafka/connector/records/RecordService.java:569]: valueschema will never be null because snowflake json converter always [returns a value schema
](https://github.com/snowflakedb/snowflake-kafka-connector/blob/master/src/main/java/com/snowflake/kafka/connector/records/SnowflakeJsonConverter.java#L31)

File: test/rest_request_template/test_string_json_ignore_tombstone.json [hunk 2]
```diff
@@ -10,8 +10,7 @@
     "snowflake.database.name":"SNOWFLAKE_DATABASE",
     "snowflake.schema.name":"SNOWFLAKE_SCHEMA",
     "key.converter":"org.apache.kafka.connect.storage.StringConverter",
-    "value.converter": "org.apache.kafka.connect.json.JsonConverter",
```
Comment (sfc-gh-rcheng) [inline test/rest_request_template/test_string_json_ignore_tombstone.json:13]: the corresponding default tombstone e2e ingestion test uses SnowflakeJsonConverter

Comment (sfc-gh-japatel) [reply test/rest_request_template/test_string_json_ignore_tombstone.json:13]: what will be the behavior if it is org.apache.kafka.connect.json.JsonConverter? Do we have a test for that?

Comment (sfc-gh-rcheng) [reply test/rest_request_template/test_string_json_ignore_tombstone.json:13]: JsonConverter treats empty string as a tombstone record whereas SnowflakeJsonConverter treats it as a normal record.

We don't have snowpipe e2e tests with JsonConverter, but we do have IT tests that [test all converters](https://github.com/snowflakedb/snowflake-kafka-connector/blob/master/src/test/java/com/snowflake/kafka/connector/internal/TombstoneRecordIngestionIT.java#L140) with snowpipe

File: test/test_suit/test_string_json.py [hunk 3]
```diff
@@ -24,7 +24,7 @@ def send(self):
                 {'numbernumbernumbernumbernumbernumbernumbernumbernumbernumbernumbernumber': str(100)}
             ).encode('utf-8'))
         else:
-            value.append('')
```
Comment (sfc-gh-tzhang) [inline test/test_suit/test_string_json.py:27]: I understand we need use None for the tombstone tests, but should we have a test to test the empty string behavior instead of updating everything to None?

**Review by sfc-gh-tzhang — APPROVED** [review]
Left one comment, otherwise LGTM!

File: test/test_suit/test_string_json.py (hunk 3 above)
Comment (sfc-gh-rcheng) [reply test/test_suit/test_string_json.py:27]: added empty string test and comment to explain the differences between community vs custom converter record behavior

File: test/rest_request_template/test_string_json_ignore_tombstone.json (hunk 2 above)
Comment (sfc-gh-japatel) [reply test/rest_request_template/test_string_json_ignore_tombstone.json:13]: Please double confirm that we dont have any behavior changes in snowpipe. Thanks

**Review by sfc-gh-japatel — APPROVED** [review]
thanks for fixing and improving test! Left a comment about being cautious in snowpipe code path.
